tests/test_ugrid.cpp

```diff
@@ -814,6 +814,306 @@ TEST( MeshUgridTest, flow_3d )
+
+  int datasetCount = static_cast<int>( timeSteps.size() );
+
+  ASSERT_EQ( datasetCount, MDAL_G_datasetCount( newGroup ) );
+
+  for ( int i = 0; i < datasetCount; ++i )
+  {
+    MDAL_DatasetH ds = MDAL_G_dataset( newGroup, i );
+    EXPECT_TRUE( compareDurationInHours( timeSteps.at( static_cast<size_t>( i ) ), MDAL_D_time( ds ) ) );
+    std::vector<double> values( valueCount );
+    MDAL_D_data( ds, 0, elemCount, isScalar ? MDAL_DataType::SCALAR_DOUBLE : MDAL_DataType::VECTOR_2D_DOUBLE, values.data() );
+
+    for ( size_t vi = 0; vi < values.size(); ++vi )
+      EXPECT_TRUE( MDAL::equals( i * 10.0 + vi * 2.0 + value, values.at( vi ), 0.001 ) );
+  }
+
+  MDAL_CloseMesh( m );
+
+  success = true;
+}
+
+TEST( MeshUgridTest, WriteDatasetNewFile )
+{
+  std::string tmpFile1 = tmp_file( "/new_file1.nc" );
+  std::string tmpFile2 = tmp_file( "/new_file2.nc" );
+  std::string tmpFile3 = tmp_file( "/new_file3.nc" );
+  std::string tmpFile4 = tmp_file( "/new_file4.nc" );
+
+  if ( fileExists( tmpFile1 ) )
+    deleteFile( tmpFile1 );
+
+  if ( fileExists( tmpFile2 ) )
+    deleteFile( tmpFile2 );
+
+  if ( fileExists( tmpFile3 ) )
+    deleteFile( tmpFile3 );
+
+  if ( fileExists( tmpFile4 ) )
+    deleteFile( tmpFile4 );
+
+  std::string baseFile = test_file( "/ugrid/D-Flow1.1/simplebox_hex7_map.nc" );
+  EXPECT_EQ( MDAL_MeshNames( baseFile.c_str() ), "Ugrid:\"" + baseFile + "\":mesh2d" );
+  MDAL_MeshH m = MDAL_LoadMesh( baseFile.c_str() );
+  ASSERT_NE( m, nullptr );
+  MDAL_Status s = MDAL_LastStatus();
+  EXPECT_EQ( MDAL_Status::None, s );
+
+  int groupCount = MDAL_M_datasetGroupCount( m );
+  ASSERT_EQ( 10, groupCount );
+
+  int faceCount = MDAL_M_faceCount( m );
+  int vertexCount = MDAL_M_vertexCount( m );
+
+  MDAL_DatasetGroupH existingGroup = MDAL_M_datasetGroup( m, 7 );
+  int datasetCount = MDAL_G_datasetCount( existingGroup );
+  std::vector<double> timeSteps( static_cast<size_t>( datasetCount ) );
+  for ( size_t i = 0; i < timeSteps.size(); ++i )
+    timeSteps[i] = MDAL_D_time( MDAL_G_dataset( existingGroup, i ) );
+  std::string refTime( MDAL_G_referenceTime( existingGroup ) );
+
+  bool success = false;
+
+  createNewDatasetGroupOnNewFile( m, tmpFile1, refTime, timeSteps, "new group scalar faces", MDAL_DataLocation::DataOnFaces, faceCount, true, 1.23,  success );
+  EXPECT_TRUE( success );
+
+  createNewDatasetGroupOnNewFile( m, tmpFile2, refTime, timeSteps, "new group scalar vertices", MDAL_DataLocation::DataOnVertices, vertexCount, true, 4.56, success );
+  EXPECT_TRUE( success );
+
+  createNewDatasetGroupOnNewFile( m, tmpFile3, refTime, timeSteps, "new group vector faces", MDAL_DataLocation::DataOnFaces, faceCount, false, 1.23, success );
+  EXPECT_TRUE( success );
+
+  createNewDatasetGroupOnNewFile( m, tmpFile4, refTime, timeSteps, "new group vector vertices", MDAL_DataLocation::DataOnVertices, vertexCount, false, 4.56, success );
+  EXPECT_TRUE( success );
+
+  MDAL_CloseMesh( m );
+
```

Comment:
would be good to load the written file and check few values/variables?

Reply:
dataset values are checked in `createNewDatasetGroupOnNewFile()`